src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -2382,6 +3275,69 @@ type Plan {
     debugOutput: debugOutput!
 }
 
+"""
+Plan (result of an itinerary search) that follows
+[GraphQL Cursor Connections Specification](https://relay.dev/graphql/connections.htm).
+"""
+type PlanConnection {
+    """
+    What was the starting point for the itinerary search.
+    """
+    searchDateTime: OffsetDateTime
+
+    """
+    Origin of the itinerary search.
+    TODO figure out the best output type
+    """
+    origin: Place!
+
+    """
+    Destination of the itinerary search.
+    TODO figure out the best output type
+    """
+    destination: Place!
```

Comment:
The issue with these places is that `Place` type should not be used since it has mandatory arrival and departure time fields. We can't use `PlaceInterface` either because it's used in the nearest places search and in the future, we want to add trip (or a stop at a trip) as a location for the plan query and the trip doesn't make sense as a potential output for the nearest search and we also need a coordinate location type. Also, we potentially could want to return both the original name of a stop for example and the label user has given for the stop (the Stop type itself doesn't have both).

What we could do is add another interface/union that is only used in this context. However, that still doesn't solve the issue that we want both the name of the location and the user given label for it. Therefore, I suggest the following schema:
```
origin: PlanLabeledLocation!

destination: PlanLabeledLocation!
```
```
type PlanLabeledLocation {  
    label: String
    
    // This can be null if for example user provides a stop id but that stop doesn't exist.
    location: PlanLocation
}
```

```
union PlanLocation =  Stop | PlanCoordinates
```

```
type PlanCoordinates {
    latitude: Coordinate!

    longitude: Coordinate!
}
```

Reply:
* The existing Coordinates type has the issue that it doesn't use the Coordinate scalar. I was a bit struggling with the naming conventions because I wanted the scalar to be called Coordinate but I think CoordinateValue is a good idea. I think I've gone back and worth with naming the coordinate `coordinate` or `coordinates` but I agree with you that it should be called `coordinate`. I think I ended up in the current naming by accident.
* I was worried the PlanLocation would be too generic as we might end up having a location without label at some point for some purpose but I think the locations within itineraries should be called ItineraryLocation or whatever so it's not as risky. I'm not sure what is the best practice with naming unions but having the "type" name as a postfix feels a bit wrong but it seems to be a convention with the input at least.

Reply:
This is a bit contrary to common language use. A coordinate (singular) is usually understood as the x or y part, and the tuple (x,y) is called coordinates (plural). So you have the latitude *coordinate* and the longitude *coordinate* and together they form the *coordinates*. 

https://www.oxfordlearnersdictionaries.com/definition/american_english/coordinate_2

Reply:
I was under the same impression as @t2gran that coordinate would be the correct term for a lat lon pair even though coordinates is used for it often but now when I try to look information about it, I can mostly just find information pointing to the other direction.